From 9c730437576fb69e842aa5bf17667e755c1c9cda Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 2 Sep 2024 19:46:23 +0200
Subject: [PATCH 1172/1695] dts: stm32: define SoC GPIO banks that are firewall
 controllers

Add property #access-controller-cells to GPIO banks that register
to the firewall framework.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6a2953a6e0336e097eea30d0fdc6f82f3a2b287f dts: stm32: define SoC GPIO banks that are firewall controllers]
Change-Id: I7c0711a3a357413e1928df1dd8aaa0361aa37715
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/401627
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi |  9 +++++++++
 core/arch/arm/dts/stm32mp151.dtsi |  1 +
 core/arch/arm/dts/stm32mp211.dtsi | 10 ++++++++++
 core/arch/arm/dts/stm32mp231.dtsi | 12 ++++++++++++
 core/arch/arm/dts/stm32mp251.dtsi | 12 ++++++++++++
 5 files changed, 44 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index 0060549ba..72f803a16 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -481,6 +481,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				clocks = <&rcc GPIOA>;
 				reg = <0x0 0x400>;
 				st,bank-name = "GPIOA";
@@ -493,6 +494,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				clocks = <&rcc GPIOB>;
 				reg = <0x1000 0x400>;
 				st,bank-name = "GPIOB";
@@ -505,6 +507,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				clocks = <&rcc GPIOC>;
 				reg = <0x2000 0x400>;
 				st,bank-name = "GPIOC";
@@ -517,6 +520,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				clocks = <&rcc GPIOD>;
 				reg = <0x3000 0x400>;
 				st,bank-name = "GPIOD";
@@ -529,6 +533,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				clocks = <&rcc GPIOE>;
 				reg = <0x4000 0x400>;
 				st,bank-name = "GPIOE";
@@ -541,6 +546,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				clocks = <&rcc GPIOF>;
 				reg = <0x5000 0x400>;
 				st,bank-name = "GPIOF";
@@ -553,6 +559,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				clocks = <&rcc GPIOG>;
 				reg = <0x6000 0x400>;
 				st,bank-name = "GPIOG";
@@ -565,6 +572,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				clocks = <&rcc GPIOH>;
 				reg = <0x7000 0x400>;
 				st,bank-name = "GPIOH";
@@ -577,6 +585,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				clocks = <&rcc GPIOI>;
 				reg = <0x8000 0x400>;
 				st,bank-name = "GPIOI";
diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index 294fcd61a..36de0fb0b 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -666,6 +666,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0 0x400>;
 				clocks = <&rcc GPIOZ>;
 				st,bank-name = "GPIOZ";
diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 5e1db084c..56ab303d1 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -805,6 +805,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x0 0x400>;
 				clocks = <&rcc CK_BUS_GPIOA>;
 				st,bank-name = "GPIOA";
@@ -816,6 +817,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x10000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOB>;
 				st,bank-name = "GPIOB";
@@ -827,6 +829,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x20000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOC>;
 				st,bank-name = "GPIOC";
@@ -838,6 +841,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x30000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOD>;
 				st,bank-name = "GPIOD";
@@ -849,6 +853,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x40000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOE>;
 				st,bank-name = "GPIOE";
@@ -860,6 +865,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x50000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOF>;
 				st,bank-name = "GPIOF";
@@ -871,6 +877,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x60000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOG>;
 				st,bank-name = "GPIOG";
@@ -882,6 +889,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x70000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOH>;
 				st,bank-name = "GPIOH";
@@ -893,6 +901,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x80000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOI>;
 				st,bank-name = "GPIOI";
@@ -964,6 +973,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0 0x400>;
 				clocks = <&rcc CK_BUS_GPIOZ>;
 				st,bank-name = "GPIOZ";
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index badbebbd3..2c1c8814a 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -989,6 +989,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x0 0x400>;
 				clocks = <&rcc CK_BUS_GPIOA>;
 				st,bank-name = "GPIOA";
@@ -1000,6 +1001,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x10000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOB>;
 				st,bank-name = "GPIOB";
@@ -1011,6 +1013,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x20000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOC>;
 				st,bank-name = "GPIOC";
@@ -1022,6 +1025,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x30000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOD>;
 				st,bank-name = "GPIOD";
@@ -1033,6 +1037,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x40000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOE>;
 				st,bank-name = "GPIOE";
@@ -1044,6 +1049,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x50000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOF>;
 				st,bank-name = "GPIOF";
@@ -1055,6 +1061,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x60000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOG>;
 				st,bank-name = "GPIOG";
@@ -1066,6 +1073,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x70000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOH>;
 				st,bank-name = "GPIOH";
@@ -1077,6 +1085,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x80000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOI>;
 				st,bank-name = "GPIOI";
@@ -1088,6 +1097,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x90000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOJ>;
 				st,bank-name = "GPIOJ";
@@ -1099,6 +1109,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0xa0000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOK>;
 				st,bank-name = "GPIOK";
@@ -1161,6 +1172,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0 0x400>;
 				clocks = <&rcc CK_BUS_GPIOZ>;
 				st,bank-name = "GPIOZ";
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 313783381..125834bee 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -1098,6 +1098,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x0 0x400>;
 				clocks = <&rcc CK_BUS_GPIOA>;
 				st,bank-name = "GPIOA";
@@ -1109,6 +1110,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x10000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOB>;
 				st,bank-name = "GPIOB";
@@ -1120,6 +1122,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x20000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOC>;
 				st,bank-name = "GPIOC";
@@ -1131,6 +1134,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x30000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOD>;
 				st,bank-name = "GPIOD";
@@ -1142,6 +1146,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x40000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOE>;
 				st,bank-name = "GPIOE";
@@ -1153,6 +1158,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x50000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOF>;
 				st,bank-name = "GPIOF";
@@ -1164,6 +1170,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x60000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOG>;
 				st,bank-name = "GPIOG";
@@ -1175,6 +1182,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x70000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOH>;
 				st,bank-name = "GPIOH";
@@ -1186,6 +1194,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x80000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOI>;
 				st,bank-name = "GPIOI";
@@ -1197,6 +1206,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0x90000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOJ>;
 				st,bank-name = "GPIOJ";
@@ -1208,6 +1218,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0xa0000 0x400>;
 				clocks = <&rcc CK_BUS_GPIOK>;
 				st,bank-name = "GPIOK";
@@ -1272,6 +1283,7 @@
 				#gpio-cells = <2>;
 				interrupt-controller;
 				#interrupt-cells = <2>;
+				#access-controller-cells = <1>;
 				reg = <0 0x400>;
 				clocks = <&rcc CK_BUS_GPIOZ>;
 				st,bank-name = "GPIOZ";
-- 
2.47.2

